### DISCUSSION TITLE
FieldResolver inheritance

### DISCUSSION THREAD

**Discussion by jpbarbosa** [MichalLytek/type-graphql#1031] [Q&A] [answered]
Following the resolver inheritance recipes and further help from @MichalLytek (https://github.com/MichalLytek/type-graphql/discussions/1029) I was able to run queries and mutations from an abstract class. Now I facing an error with FieldResolver:

```ts
export const createRecordResolver = <T extends Record>(
  objectTypeCls: ClassType<T>
) => {
  @Resolver({ isAbstract: true })
  abstract class BaseResolver {
    @FieldResolver()
    async user(@Root() record: T) {
      console.log(record);
    }
  }

  return BaseResolver;
};
```

```bash
Error: No provided object type in '@Resolver' decorator for class 'BaseResolver!'
```

The same situation was previously mentioned in this issue: https://github.com/MichalLytek/type-graphql/issues/102

It would be great to have some FieldResolver examples in the documentation.

Thank you again.

**Comment by MichalLytek** [comment, marked as answer]
```ts
export const createRecordResolver = <T extends Record>(
  objectTypeCls: ClassType<T>
) => {
  @Resolver(() => objectTypeCls, { isAbstract: true })
  abstract class BaseResolver {
    @FieldResolver()
    async user(@Root() record: T) {
      console.log(record);
    }
  }

  return BaseResolver;
};
```

When you understand the basic TypeGraphQL classes and `() => ` metadata, then you will learn how to create any type of types inheritance.

**Reply by jpbarbosa** [reply]
Wow, it worked. Good to see TypeGraphQL inheritance works very well. It avoids a lot of boilerplate code.

And, yes, I need to put more effort on the basics. Thank you!
